scripts/ci/ws12_workflow_contracts.py

```
@@ -136,7 +147,327 @@ def validate_e2e_scope_filters(text: str) -> list[str]:
+            return None, detail
+        return (lambda path: bool(pattern.search(path))), None
+    globs, detail = extract_paths_globs(text)
+    if detail is not None:
+        return None, detail
+    compiled = [github_glob_to_regex(glob) for glob in globs]
+    return (
+        lambda path: any(pattern.match(path) for pattern in compiled)
+    ), None
+
+
+def validate_crate_scope_filters(
```

Ruff (0.16.0)

[warning] Too many branches (14 > 12)

(PLR0912)